Documentation/Contributing.md

```diff
@@ -0,0 +1,350 @@
+8. Push the commit / branch to your fork
+9. Make a PR from your fork / branch to `apple: main`
+10. While creating your PR, make sure to add one of the [PR Templates](https://github.com/apple/swift-package-manager/tree/main/.github) in your PR message depending on the reasoning of your PR.
+11. Reviewers are going to be automatically added to your PR
+12. Merge pull request when you received approval from the reviewers (one or more)
+
+## Using Continuous Integration
+SwiftPM uses [swift-ci](https://ci.swift.org) infrastructure for its continuous integration testing. The bots can be triggered on pull-requests if you have commit access. Otherwise, ask
+one of the code owners to trigger them for you. The following commands are supported:
+
+```
+@swift-ci please smoke test
+```
+
+Run tests with the trunk compiler and other projects. This is **required** before
+a pull-request can be merged.
+
+```
+@swift-ci please smoke test self hosted
+```
+
+Run just the self-hosted tests. This has fast turnaround times so it can be used
+to get quick feedback.
+
+Note: Smoke tests are still required for merging pull-requests.
+
+## Using Custom Swift Compilers
+SwiftPM needs the Swift compiler to parse `Package.swift` manifest files and to
+compile Swift source files. You can use the `SWIFT_EXEC` and `SWIFT_EXEC_MANIFEST`
+environment variables to control which compiler to use for these operations.
+
+`SWIFT_EXEC_MANIFEST`: This variable controls which compiler to use for parsing
+`Package.swift` manifest files. The lookup order for the manifest compiler is:
+`SWIFT_EXEC_MANIFEST`, `swiftc` adjacent to the `swiftpm` binaries, then `SWIFT_EXEC`
+
+`SWIFT_EXEC`: This variable controls which compiler to use for compiling Swift
+sources. The lookup order for the sources' compiler is: `SWIFT_EXEC`, then `swiftc` adjacent
+to `swiftpm` binaries. This is also useful for Swift compiler developers when they
+want to use a debug compiler with SwiftPM.
+
+```bash
+$> SWIFT_EXEC=/path/to/my/built/swiftc swift build
+```
+
+## Overriding the Path to the Runtime Libraries
+SwiftPM computes the path of its runtime libraries relative to where it is
+installed. This path can be overridden by setting the environment variable
+`SWIFTPM_PD_LIBS` to a directory containing the libraries, or a colon-separated list of
+absolute search paths. SwiftPM will choose the first
+path which exists on disk. If none of the paths are present on disk, it will fall
+back to built-in computation.
+
+## Making changes in TSC targets
+All targets with the prefix TSC define the interface for the tools support core. Those APIs might be used in other projects as well and need to be updated in this repository by copying their sources directories to the TSC repository. The repository can be found [here](https://github.com/apple/swift-tools-support-core).
```

Comment:
SwiftPM uses [Tools Support Core](https://github.com/apple/swift-tools-support-core) (aka TSC) for many of its general purpose utilities. Changes in SwiftPM often require changes in TSC first. To coordinate changes, open a PR against TSC first, then a second one against SwiftPM pulling the correct TSC version.